File: code/src/MarcheGrosServer/Orders/Order.java
package MarcheGrosServer.Orders;

import TrackingCode.CountryEnum;
import TrackingCode.ExtractModeEnum;
import TrackingCode.TypeEnergyEnum;

import org.json.JSONObject;

public class Order{
    private TypeEnergyEnum typeEnergy;
    private CountryEnum countryOrigin;
    private ExtractModeEnum extractionMode; 
    private boolean greenEnergy; 
    private int quantity;
    private int quantityMin; 
    private int budget; 
    private int maxPriceUnitEnergy;


    public Order(TypeEnergyEnum typeEnergy, CountryEnum countryOrigin, ExtractModeEnum extractionMode, boolean greenEnergy, 
            int quantity, int quantityMin, int budget, int maxPriceUnitEnergy){
        this.typeEnergy = typeEnergy;
        this.countryOrigin = countryOrigin;
        this.extractionMode = extractionMode;
        this.greenEnergy = greenEnergy;
        this.quantity = quantity;
        this.quantityMin = quantityMin;
        this.budget = budget;
        this.maxPriceUnitEnergy = maxPriceUnitEnergy;
    }

    public Order fromJSON(JSONObject orderJSON){
        return new Order(
            TypeEnergyEnum.valueOf(orderJSON.getString("typeEnergy")),
            CountryEnum.valueOf(orderJSON.getString("countryOrigin")),
            ExtractModeEnum.valueOf(orderJSON.getString("extractionMode")),
            Boolean.valueOf(orderJSON.getBoolean("greenEnergy")),
            Integer.valueOf(orderJSON.getString("quantity")),
			Integer.valueOf(orderJSON.getString("quantityMin")),
			Integer.valueOf(orderJSON.getString("budget")),
			Integer.valueOf(orderJSON.getString("maxPriceUnitEnergy"))
        );
    }

    public JSONObject toJSON(){
        JSONObject orderJSON = new JSONObject();
        orderJSON.put("typeEnergy", this.typeEnergy);
        orderJSON.put("countryOrigin", this.countryOrigin);
        orderJSON.put("extractionMode", this.extractionMode);
        orderJSON.put("greenEnergy", this.greenEnergy);
        orderJSON.put("quantity", this.quantity);
        orderJSON.put("quantityMin", this.quantityMin);
        orderJSON.put("budget", this.budget);
        orderJSON.put("maxPriceUnitEnergy", this.maxPriceUnitEnergy);
        return orderJSON;
    }

    public TypeEnergyEnum getTypeEnergy(){
        return this.typeEnergy;
    }

    public CountryEnum getCountryOrigin(){
        return this.countryOrigin;
    }

    public ExtractModeEnum getExtractionMode(){
        return this.extractionMode;
    }

    public boolean getGreenEnergy(){
        return this.greenEnergy;
    }

    public int getQuantity(){
        return this.quantity;
    }

    public int getQuantityMin(){
        return this.quantityMin;
    }

    public int getBudget(){
        return this.budget;
    }

    public int getMaxPriceUnitEnergy(){
        return this.maxPriceUnitEnergy;
    }
    
}

File: code/php/app/models/Order.php
<?php
/**
 * ***********************************************************************************************************
 * Model gérant une commande dans sa globalité implémentant la classe JsonSerializable
 * @author HADID Hocine & CHEMIN Pierre
 * @version 1.0
 * ***********************************************************************************************************
 */

class Order implements JsonSerializable{
    // Attributs privés
    private Client $client;
    private Energy $energy;
    private float $quantity;
    private float $minQuantity; 
    private float $maxUnitPrice;
    private String $originCountry;
    private float $budget; 
    private int $portTare;
    
    /**
     * Constructeur par initialisation
     *
     * @param Client $client
     * @param Energy $energy
     * @param float $quantity
     * @param float $minQuantity
     * @param float $maxUnitPrice
     * @param String $originCountry
     * @param float $budget
     * @param String $status
     */
    public function __construct(Client $client, Energy $energy, float $quantity, float $minQuantity, float $maxUnitPrice, 
                                String $originCountry, float $budget){
        $this->client = $client;
        $this->energy = $energy;
        $this->quantity = $quantity;
        $this->minQuantity = $minQuantity;
        $this->maxUnitPrice = $maxUnitPrice;
        $this->originCountry = $originCountry;
        $this->budget = $budget;
    }

    /**
     * Méthode permettant de sérialiser l'objet en JSON
     * @return array
     */
    public function jsonSerialize():array{
        return [
            'name' => $this->client->getName(),
            'surname' => $this->client->getSurname(),
            'email' => $this->client->getMail(),
            'phoneNumber' => $this->client->getPhoneNumber(),
            'companyName' => $this->client->getNameCompany(),
            'typeEnergy' => $this->energy->getTypeEnergy(),
            'countryOrigin' => $this->originCountry,
            'extractionMode' => $this->energy->getExtractionMode(),
            'green' => $this->energy->getGreenEnergy(),
            'quantity' => $this->quantity,
            'quantityMin' => $this->minQuantity,
            'budget' => $this->budget,
            'maxPriceUnitEnergy' => $this->maxUnitPrice,
        ];
    }

    /**
     * Récupération du client
     * @return Client
     */
    public function getClient(){
        return $this->client;
    }

    /**
     * Récupération de l'énergie
     * @return Energy
     */
    public function getEnergy(){
        return $this->energy;
    }

    public function getQuantity(){
        return $this->quantity;
    }

    public function getMinQuantity(){
        return $this->minQuantity;
    }

    public function getMaxUnitPrice(){
        return $this->maxUnitPrice;
    }

    public function getOriginCountry(){
        return $this->originCountry;
    }

    public function getBudget(){
        return $this->budget;
    }
    
}